### DISCUSSION TITLE
Next/Previous Split While Zoomed without Un-Zooming

### DISCUSSION THREAD

**Discussion by bowmanmike** [ghostty-org/ghostty#3484] [Feature Requests, Ideas] [answered]
In iTerm, you can toggle between the next/previous zoomed pane without un-zooming, whereas in Ghostty if you swap between next/previous pane while zoomed, it un-zooms. Screen recording of iTerm behaviour attached here

https://github.com/user-attachments/assets/16a5b34b-7e33-45f5-9359-ef7d6ca5a5cd

**Comment by mitchellh** [comment]
This is done on purpose. When we designed the zoom split feature we discussed and landed on this as preferred for us. I'm wiling to look back into it if there is a lot more request for this but for now I'm going to mark this as rejected.

**Reply by bowmanmike** [reply]
Gotcha, appreciate the response! Would love the ability to toggle the behaviour but understand that it's a low priority issue. Thanks!

**Reply by elamaranae** [reply]
Would be interested to contribute if adding a non-default configuration is acceptable.

Thank you anyways.

**Reply by jamesofficer** [reply]
I'd also like to see this as a configurable option. I find having to re-zoom into the pane each time a little tiresome when I want to quickly jump between panes.

**Reply by michix** [reply]
+1

See #5052 .

**Reply by poige** [reply]
++

**Reply by danieluhl** [reply]
This is the only reason I haven't switched to ghostty from kitty. I'm really curious why someone would want to change the layout automatically while switching panes. I get the feeling that most people who don't complain about this are still using tmux.

**Reply by poige** [reply]
> This is the only reason I haven't switched to ghostty from kitty.

Not that anyone cares. Ironically, I still run Ghostty more often — simply because its dev doesn't treat "bold as brightness" like a mortal sin. There's an option for it. In Kitty? Nope — it's a holy cow. Tradition, lol.

Still, iTerm leaves them all in the dust. My condolences to the non-macOS crowd. ;)

**Comment by rhodes-b** [comment, marked as answer]
This is already a accepted feature request 
https://github.com/ghostty-org/ghostty/issues/8458
